### ISSUE TITLE
Arrow function methods not recognised

### ISSUE THREAD

**Issue by testpossessed** [aljazsim/vs-code-typescript-class-organizer#16]
Thanks for the feature to support control of member ordering.  Works great, but I have found one issue that spoils it a bit.

If I define a member as an arrow function, as needed to maintain this context when creating event handlers etc, it is not recognised as a method instead it is organised as a property.  This causes a violation of tslint member-order rule with the following message:

_Declaration of public instance field not allowed after declaration of public instance method. Instead, this should come at the beginning of the class/interface_

I had a look at the source code and it looks like the problem might be with TypeScript not recognising the member as a method.  So not sure if you can do anything about it but would appreciate you taking a look.

I have attached a screen shot to help understand the problem. 
 
![image](https://user-images.githubusercontent.com/10528120/71874845-d9ba5700-311a-11ea-94d6-17bc06a60b66.png)

**Comment by aljazsim** [comment]
Hmm. These indeed look like properties. A small suggestion you could do to refactor your code to achieve the desired result:

1. Instead of using arrow functions create actual class methods.
2. Assign the method reference to the event handler property (as a default event handler I assume). 

Example:

public eventHandler = this.defaultEventHandler;

public defaultEventHandler() {
  ...
}

**Comment by testpossessed** [comment]
Thanks for the response.

I am not sure that would work.  The reason these particular methods are defined as arrow functions is so that TypeScript maintains the 'this' context as the class when the method is triggered by an external event such as a click on a button.  There are several class methods in that file but if I make these ones class methods then when they are called via an Angular binding  'this' is not the class.  I'm no TypeScript guru but I am not sure how the refactoring would fix that.

Also I'm not sure the rest of the team would want to change our coding style to suit an extension I chose to use.  I will use the extension to do most of the organisation then manually finish it off.

It would be nice if the extension recognised these as methods though or had an option to do so, TSLint recognises them as methods and it is a common pattern to maintan the 'this' context.

**Comment by aljazsim** [comment]
I understand. JavaScript and this sope is always fun :). I'll see if I can do something about it.

**Comment by testpossessed** [comment]
Thanks that would be great.  In the middle of a massive refurb of our house at the moment otherwise I would normally try to find a solution and submit a pull request.

**Comment by testpossessed** [comment]
I have a bit of time to work on a PR for this issue.  I have forked and cloned the repo and found what I think is a solution, but for some reason I am finding it hard to debug the project.

I added a launch configuration based on the one generated for a new extension project, and it seems to run, but the extension host doesn't actually launch I just get the debug bar and nothing else.  How do you debug?

**Comment by testpossessed** [comment]
I tried to create a PR with a fix, but the root of the problem is TypeScript itself.

I tried to add a check to transformer.ts so that it recognises the member as an arrow function using ts.isArrowFunction but that did not work.  On debugging the code I found the problem is that TypeScript itself does not recognise the arrow function instead it sets the kind property of the node representing the arrow function to SyntaxKind.PropertyDeclaration instead of SyntaxKind.ArrowFunction.  So nothing we can do for now.

I have posted an GitHub issue in the TypeScript repo https://github.com/microsoft/TypeScript/issues/37247
